### PR TITLE
python312Packages.skops: init at 0.10

### PR DESCRIPTION
**Pull request by ViZiD**
## Description of changes
skops is a Python library helping you share your scikit-learn based models and put them in production
https://github.com/skops-dev/skops

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/skops/default.nix [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    rich
+    catboost
+    xgboost
+    lightgbm
+    pandas
+    matplotlib
+    flaky
+    quantile-forest
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/skops/default.nix:70]:
```suggestion
  optional-dependencies = {
    rich = [ rich ];
  };

  nativeCheckInputs = [
    pytestCheckHook
    pythonRelaxDepsHook

    catboost
    xgboost
    lightgbm
    pandas
    matplotlib
    flaky
    quantile-forest
  ] ++ optional-dependencies.rich;
```
https://github.com/skops-dev/skops/blob/9b565a0672a8c25c8eac38604a07cdcf7047f757/setup.py#L78-L82

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 307515` at 83b52b5f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python311Packages.skops</li>
    <li>python312Packages.skops</li>
  </ul>
</details>

**Comment by ViZiD** [comment]
> Result of `nixpkgs-review pr 307515` at [83b52b5](https://github.com/NixOS/nixpkgs/commit/83b52b5fcca727d8cf3daa538e3eec71054218e7) run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages built successfully:
> Result of `nixpkgs-review pr 307515` at [83b52b5](https://github.com/NixOS/nixpkgs/commit/83b52b5fcca727d8cf3daa538e3eec71054218e7) run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages failed to build:

i don't have devices on that platform, i mark package broken on that platform for now

**Comment by ViZiD** [comment]
@Aleksanaa can you review pls?

**Comment by ViZiD** [comment]
https://github.com/NixOS/nixpkgs/pull/317010

**Pull request closed without merging**